moodler/sections.py

```diff
@@ -152,55 +153,49 @@ def get_assignments_by_section(course_id, sections_names=None, assignments_names
     """
     Retrieving assignments by sections.
     """
-    exercises_by_sections = {}
-    sections_not_found = []
-    assignments_not_found = []
-
     # Retrieve the contents of the course
     sections = core_course_get_contents(course_id)
 
     if sections_names is not None:
-        sections_not_found = sections_names[:]
+        # Filter out section names
+        sections = [section for section in sections if section["name"] not in sections_names]
+
+        found_sections = set(section["name"] for section in sections)
+        missing_sections = set(sections_names).difference(found_sections)
+
+        if missing_sections:
+            logger.error("Could not find the following sections: %s", list(missing_sections))
 
     if assignments_names is not None:
-        assignments_not_found = assignments_names[:]
+        # Handle missing assignments
+        found_assignments = set(
+            [module["name"] for section in sections for module in section["modules"]]
+        )
 
-    # Looking through the course contents trying to locate the assignment
-    for section in sections:
-        # Making sure this section is one of the sections we are supposed to
-        # be looking at, unless the list is None
-        if sections_names is not None:
-            if section["name"] not in sections_names:
-                continue
+        missing_assignments = set(assignments_names).difference(found_assignments)
+        if missing_assignments:
+            logger.error("Could not find the following assignments: %s", missing_assignments)
 
-            sections_not_found.remove(section["name"])
+    assignment_id_to_section = defaultdict(list)
```

Comment:
Shouldn't this return `str` by default?
I got a little lost here
I understand why `assignments_by_section` is list by default, but why `assignment_id_to_section` is list I didn't understand